File: src/main/java/top/dsbbs2/bukkitcord/api/IPlugin.java
package top.dsbbs2.bukkitcord.api;

import java.io.File;

public interface IPlugin extends ICommandExecutor, ITabCompleter {
    void onLoad();
    void onEnable();
    void onDisable();
    File getDataFolder();
    IPluginDescription getPlatformDescription();
    Set<IListener> getListeners();
    boolean isEnabled();
}

File: src/main/java/top/dsbbs2/bukkitcord/api/IPlugin.kt
package top.dsbbs2.bukkitcord.api

import java.io.File

interface IPlugin : ICommandExecutor, ITabCompleter {
    fun onLoad()
    fun onEnable()
    fun onDisable()
    val dataFolder: File?
    val platformDescription: IPluginDescription?
    val listeners: Set<IListener?>?
    val isEnabled: Boolean
}